src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.MetadataDb.cs

```diff
@@ -149,7 +149,7 @@ internal MetadataDb(MetadataDb source, bool useArrayPools)
 
             public void Dispose()
             {
-                byte[] data = Interlocked.Exchange(ref _data, null);
+                byte[]? data = Interlocked.Exchange(ref _data, null!);
```

Comment:
nit: consider making `_data` nullable and adding `Debug.Assert(_data != null)` to the `TrimExcess()`, `Append`, `Enlarge` methods, rather than using `null!` here.

@bartonjs - thoughts?

Reply:
I agree.  `_data` is nullable.

Reply:
Normally we don't make a field nullable when it is always initiated in constructors and only nulled/cleared in Dispose(). For me it is an extra burden, unless you guys suspect it might be used somehow after dispose

Reply:
If that's the model used elsewhere, it's fine I guess.  It's technically wrong, but not enough to argue.

Reply:
> Normally we don't make a field nullable when it is always initiated in constructors and only nulled/cleared in Dispose()

Is there a guidance written somewhere about this or precedence you can point me to as an example for such decisions? Just asking for my edification :)

Reply:
Yes, but here mentioned only properties because that is more important as it is exposed. Same should go for fields, we haven't documented that though https://github.com/dotnet/apireviews/tree/master/2019/System.Private.CoreLib-Nullable

![image](https://user-images.githubusercontent.com/45579687/70183364-7f9d2080-169a-11ea-8e82-49aad7de4f7f.png)

Reply:
This is also discussed in https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/api-guidelines/nullability.md#annotation-guidance.  If using an instance after it's been disposed is a violation of its contract (which is almost always the case with .NET types), then it's contract (including those specified by nullable annotations) effectively go out the window, so we don't harm the correct usage case by catering to the incorrect usage case.